### PR TITLE
bash_ops

### PR DESCRIPTION
**Pull request by harou24**

### CODE REVIEW

**Comment by harou24** [comment]
how to run : 
 ./minimake.sh test && ./build/apps/builtin_echo.bin echo hello world "test                     one two"

**Comment by harou24** [comment]
Run with : 
./minimake.sh test && ./build/apps/builtin_echo.bin echo command

File: src/builtin_echo.c [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		count = 2;
+	else
+		count = 1;
+	while (count < cmd->argv->argc)
+	{
+		ft_dprintf(STDERR, "%s", cmd->argv->argv[count]);
+		if (count != cmd->argv->argc - 1)
+			ft_dprintf(STDERR, " ");
+		if (!is_n_flag(cmd))
+			ft_dprintf(STDERR, "\n");
```
Comment (s-t-a-n) [inline src/builtin_echo.c]: this will not work in a pipe, check includes/command.h

File: src/builtin_echo.c [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		count++;
+	}
+	return (1);
+}
```
Comment (s-t-a-n) [inline src/builtin_echo.c:38]: idea is fine, fix the bugs, and you can optimize this and write it a lot shorter. also norminette!

File: src/builtin_echo.c [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#include "bash_ops.h"
+#include "env_singleton.h"
+#include "ft_printf.h"
+#include "libft.h"
+
+int	is_n_flag(t_command *cmd)
+{
+	if (ft_strcmp(cmd->argv->argv[1], "-n") == 0)
+		return 1;
+	return (0);
```
Comment (s-t-a-n) [inline src/builtin_echo.c]: can you think of a condition where this will SEGV?

File: src/builtin_echo.c [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+int	builtin_echo(t_command *cmd)
+{
+	int	count;
+
+	if (is_n_flag(cmd))
```
Comment (s-t-a-n) [inline src/builtin_echo.c]: why not use a proper bool? you call is_n_flag again, therefore calling ft_strcmp twice.

File: src/builtin_echo.c [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#include "bash_ops.h"
+#include "env_singleton.h"
+#include "ft_printf.h"
+#include "libft.h"
+
+int	is_n_flag(t_command *cmd)
```
Comment (s-t-a-n) [inline src/builtin_echo.c]: make static, is_...() should return bool

File: src/builtin_echo.c [hunk 6]
```diff
@@ -0,0 +1,31 @@
+#include "bash_ops.h"
+#include "env_singleton.h"
```
Comment (s-t-a-n) [inline src/builtin_echo.c]: echo doesnt use environment right? it doesnt matter, but why include it

File: src/builtin_echo.c [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+	while (count < cmd->argv->argc)
+	{
+		ft_dprintf(STDERR, "%s", cmd->argv->argv[count]);
+		if (count != cmd->argv->argc - 1)
+			ft_dprintf(STDERR, " ");
+		if (!is_n_flag(cmd))
+			ft_dprintf(STDERR, "\n");
+		count++;
+	}
+	return (1);
```
Comment (s-t-a-n) [inline src/builtin_echo.c]: return 1? bash functions return 0 on success

**Review by s-t-a-n — CHANGES_REQUESTED** [review]
idea is fine, fix the bugs, and you can optimize this and write it a lot shorter. also norminette!

**Review by s-t-a-n — APPROVED** [review]

**Pull request merged**